apps/protocol-frontend/src/components/DaoDashboardShell.tsx

```diff
@@ -1,24 +1,110 @@
+import { useState } from 'react';
 import { Box, Flex, Heading } from '@chakra-ui/react';
+import { ControlledSelect, ControlledDatePicker } from '@govrn/protocol-ui';
 import PageHeading from '../components/PageHeading';
 import ContributionTypesPieShell from './ContributionTypesPieShell';
 import RecentContributionsTableShell from './RecentContributionsTableShell';
+
+import { subWeeks } from 'date-fns';
 interface DaoDashboardShellProps {
   daoName: string;
   daoId: number;
 }
-
-const startDate = new Date('7/1/2022'); // temporary hardcoded value while working on the datepicker component
-const endDate = new Date('8/1/2022'); // temporary hardcoded value while working on the datepicker component
+const TODAY_DATE = new Date();
 
 const DaoDashboardShell = ({ daoName, daoId }: DaoDashboardShellProps) => {
+  const [showCustomDatePicker, setShowCustomDatePicker] = useState(false);
+
+  const [startDate, setStartDate] = useState<Date>(new Date(TODAY_DATE));
+  const [endDate, setEndDate] = useState<Date>(
+    new Date(subWeeks(TODAY_DATE, 4)),
+  );
+  const [dateRange, setDateRange] = useState<{
+    label: string;
+    value: number;
+  } | null>(null);
+
+  const dateRangeOptions = [
+    {
+      value: 'Custom',
+      label: 'Custom',
+    },
+    { value: 1, label: 'Last Week' },
+    { value: 4, label: 'Last Month' },
+    { value: 12, label: 'Last Quarter' },
+    { value: 52, label: 'Last Year' },
```

Comment:
It may be good to move these number to name consts. We could probably reuse them in the UserDashboard to make it clearer what is going on in the code.

Reply:
Great idea, will help clean things up and also make it so we can more directly update this if we want to change our pre-defined date values